src/features/pass-survey/model/hooks/useUserActionManager.ts

```diff
@@ -43,6 +44,7 @@ function useUserActionManager({ activityId, activityState }: Args) {
     const shouldUpdateLastAction =
       lastUserAction &&
       lastUserAction?.payload?.activityItemId === currentPipelineItem.id &&
+      lastUserAction.type === 'SET_ANSWER' &&
```

Comment:
Why would we need this check?
updateUserActionsWithAnswer implies to be called only SET_ANSWER action

Reply:
Because AB Trails has 4 ab-trail activity items and 4 tutorials. Every totorial has the same id as the next ab-trail item. So, we need to update only ab-trail and do not affect tutorial.